projects/angular/src/data/datagrid/datagrid-detail.ts

```diff
@@ -38,8 +39,32 @@ import { DetailService } from './providers/detail.service';
 export class ClrDatagridDetail {
   @ContentChild(ClrDatagridDetailHeader) header: ClrDatagridDetailHeader;
 
+  @Input('clrDetailAriaLabelledBy') ariaLabelledBy: string;
+  @Input('clrDetailAriaLabel') ariaLabel: string;
+
   constructor(public detailService: DetailService, public commonStrings: ClrCommonStringsService) {}
 
+  get labelledBy(): string {
+    if (this.ariaLabelledBy) {
+      return this.header ? `${this.header.titleId} ${this.ariaLabelledBy}` : this.ariaLabelledBy;
+    } else {
+      // If aria-label is set by the end user, do not set aria-labelledby
+      if (this.ariaLabel) {
+        return null;
+      } else {
+        return this.header?.titleId || '';
+      }
+    }
```

Comment:
```suggestion
    } else if (this.ariaLabel) {
      // If aria-label is set by the end user, do not set aria-labelledby.
      return null;
    } else {
      return this.header?.titleId || '';
    }
```

This can be flattened.

Reply:
Changed as suggested.